packages/rich-text-from-markdown/src/__test__/real-world.md

```diff
@@ -70,3 +70,16 @@ Inline `code`
 [link text](https://www.contentful.com)
 
 [link with title](https://www.contentful.com/blog/ "title text!")
+
+## Tables
+
+| Name       | Country |
+| ---------- | ------- |
+| Contentful | Germany |
+| Nasa       | USA     |
+
+## Tables with marks
+
+| **Bold Header 1** | **Bold Header 2** |
+| ----------------- | ----------------- |
+| _Italic_          | `Code`            |
```

Comment:
How about some more complex test cases?

1. HTML syntax
2. Hyperlinks inside a table
3. Multiple paragraphs of text inside a table (not sure if MD will actually create paragraphs inside the cells, but if so, this could clash with our logic - see [comment below](https://github.com/contentful/rich-text/pull/284/files#r740070835))
3. Some elements unsupported inside tables (e.g. blockquote, lists)
4. Colspans/Rowspans handling

Reply:
There's some limitation for what the parser returns. I did some of those tests manually to understand what it returns.

1. HTML syntax - we ignore it, the parser returns it but we filter it out
2. It returns it as a `text` - I can write tests for it
3. Multiple paragraphs - It returns an array of `text`, I tested `text 1<br />text 2`, it returns `text 1` and `text 2` as an array of `text`
4. It returns them as a `text`
5. Not sure how I can test that with markdown 🤔

Reply:
> 1. HTML syntax - we ignore it, the parser returns it but we filter it out

Only for tables or in general?
In your new tests I see e.g. `<p></p>` seems to work. If we can make it work for tables without much additional effort then I think we should. We don't have any data on it, but I'd assume a big portion of tables being written in HTML due to the complex syntax and because some things are only possible with HTML style tables in MD (no native support for things like colspan/rowspan)

> 5. Not sure how I can test that with markdown 

AFAIK it only works with HTML syntax which brings us back to the first point.

> 2. It returns it as a text - I can write tests for it

It's a hyperlink though in your test :D which is great.

Reply:
`<p>` is also ignored. The markdown parser returns a node type called `text` for the text inside the paragraph, we use it only this node then we manually wrap every text inside a paragraph. The parser also returns a node type `html` containing the `p` tag but we ignore it. We ignore all html tags, the parser returns them but we do nothing with them.

Reply:
> We ignore all html tags, the parser returns them but we do nothing with them.

I see. Could be a nice feature improvement I guess.